projects/storefrontlib/cms-components/navigation/search-box/search-box.component.ts

```diff
@@ -302,28 +343,80 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
   }
 
   // Return result list as HTMLElement array
-  private getResultElements(): HTMLElement[] {
+  protected getResultElements(): HTMLElement[] {
     return Array.from(
       this.winRef.document.querySelectorAll(
-        '.products > li a, .suggestions > li a, .recent-searches > li a'
+        '.products ul:not(.hidden) > li a, .suggestions ul  > li a, .recent-searches ul > li a,.trending-searches ul > li a, .carousel-panel .item.active > a, .products .carousel-panel > button:not([disabled])'
       )
     );
   }
 
+  // Return group list as HTMLElement array
+  private getGroupElements(): HTMLElement[][] {
+    const groups: HTMLElement[][] = [];
+    groups.push(
+      Array.from(
+        this.winRef.document.querySelectorAll(
+          '.products ul:not(.hidden) > li a'
+        )
+      )
+    );
+    groups.push(
+      Array.from(
+        this.winRef.document.querySelectorAll('.suggestions ul  > li a')
+      )
+    );
+    groups.push(
+      Array.from(
+        this.winRef.document.querySelectorAll(
+          '.trending-searches-container.d-block .trending-searches ul > li a'
+        )
+      )
+    );
+    groups.push(
+      Array.from(
+        this.winRef.document.querySelectorAll('.recent-searches ul > li a')
+      )
+    );
+
+    groups.push(
+      Array.from(
+        this.winRef.document.querySelectorAll(
+          '.carousel-panel .item.active > a, .carousel-panel > button:not([disabled])'
+        )
+      )
+    );
+    groups.push(
+      Array.from(
+        this.winRef.document.querySelectorAll('.search-panel-close-btn')
+      )
+    );
+    return groups.filter((group) => group.length);
+  }
   // Return focused element as HTMLElement
-  private getFocusedElement(): HTMLElement {
+  protected getFocusedElement(): HTMLElement {
     return <HTMLElement>this.winRef.document.activeElement;
   }
 
   updateChosenWord(chosenWord: string): void {
     this.chosenWord = chosenWord;
   }
 
-  private getFocusedIndex(): number {
+  protected getFocusedIndex(): number {
     return this.getResultElements().indexOf(this.getFocusedElement());
   }
 
-  private propagateEvent(event: KeyboardEvent) {
+  protected getFocusedGroupIndex(): number {
+    let indexGroup = 0;
+    this.getGroupElements().forEach((group, index) => {
+      if (group.indexOf(this.getFocusedElement()) !== -1) {
+        indexGroup = index;
+      }
+    });
+    return indexGroup;
```

Comment:
not very good from performance side. forEach will continue iteration till the end. 
did you consider `this.getGroupElements().findIndex(group => group.indexOf(focusedElement)?`

Reply:
👍 -> fixed